src/ChainSafe.Gaming/Web3/Evm/Wallet/WalletSigner.cs

```diff
@@ -30,7 +30,7 @@ public virtual async ValueTask WillStartAsync()
 
             PublicAddress = address.AssertIsPublicAddress();
 
-            await walletProvider.AddNetworkIfNotExistInWallet();
+            await walletProvider.SwitchChainAddIfMissing();
```

Comment:
this is already implemented in `EthereumWindow.jslib` Connect method

Reply:
I just renamed the method here